src/htm/utils/Topology.cpp

```diff
@@ -227,22 +227,39 @@ bool WrappingNeighborhood::Iterator::operator!=(const Iterator &other) const {
 
 UInt WrappingNeighborhood::Iterator::operator*() const {
   UInt index = 0;
-  for (size_t i = 0; i < neighborhood_.dimensions_.size(); i++) {
+  for (size_t i = 0; i < neighborhood_.ndimensions_; i++) {
     Int coordinate = neighborhood_.centerPosition_[i] + offset_[i];
+    const UInt a = neighborhood_.dimensions_[i]; // the compiler doesn't seem to voluntarily hold on to this value
+
 
     // With a large radius, it may have wrapped around multiple times, so use
     // `while`, not `if`.
 
-    while (coordinate < 0) {
-      coordinate += neighborhood_.dimensions_[i];
-    }
 
-    while (coordinate >= (Int)neighborhood_.dimensions_[i]) {
-      coordinate -= neighborhood_.dimensions_[i];
-    }
+//    while (coordinate < 0) {
+//      coordinate += neighborhood_.dimensions_[i];
+//    }
 
-    index *= neighborhood_.dimensions_[i];
+
+//    while (coordinate >= (Int)neighborhood_.dimensions_[i]) {
+
+
+ //     coordinate -= neighborhood_.dimensions_[i];
+
+
+ //   }
+
+
+    coordinate = coordinate % (Int) a; 
+// C++, depending on the version of the STD, may produce negative "mod n" values.
+    coordinate += ((UInt) coordinate >> shftInt) * (Int) a; // both this line and the one commented out below produced the same number of assembly instructions on my system, however this line was slightly faster for whatever reason (the instructions produced for each line were different)
+//    coordinate += (coordinate < 0) * (Int) a;
```

Comment:
again, in that case the results are system, std, platform,... dependant, maybe hold on to the more readable one

Reply:
> again, in that case the results are system, std, platform,... dependant, maybe hold on to the more readable one

I figured this one would be the safest bitwise operation. I'll change it though, no problem.